### ISSUE TITLE
Past jobs that were not completed

### ISSUE THREAD

**Issue by advancedeng** [MISP/MISP#669]
Hi all,

When I go to Administrayion->Jobs I can see some past jobs that were not completed. 
These jobs can be email that were not delivered (due to misconfigured settings) or pulling data from other MISP servers. How can I delete these jobs?

Cheers,
Andreas

**Comment by adulau** [comment]
A redis "flushdb" would do the job. @iglocska Is this the appropriate way to clean-up the jobs?

**Comment by iglocska** [comment]
There are 2 places where the jobs are stored:

1. in redis - these are the actual ongoing jobs, they have 2 states basically: scheduled to be executed at a given time (will be picked up by the scheduler worker), and executed asap (picked up on the next poll from the appropriate worker). The moment the job is picked up for execution, it is popped from the stack and removed from redis.
2. In MYSQL, in the jobs table - think of this as a secondary log table that also shows the progress of ongoing jobs. If a job fails, it will get "stuck" on the current progress. However, these are informational only and there is no need to remove these, other than for cleanup purposes. If you really want to remove them, simply log into MYSQL and run the following query:
delete from `jobs`;

What could be more interesting is to restart a worker if it gets stuck, you should be able to do this from the workers tab in the admin interface. If all else fails, flushdb is a the nuclear option that will remove all worker queues, effectively killing the workers - but it should only be used if all else fails. Also, watch out if you use ZMQ as that will also flush that data. 

tl;dr: The jobs in administration->jobs are informational only, no need to remove them. If a job fails it's gone for good.

**Comment by elhoim** [comment]
Answered, so closing it.
